File: java/src/test/java/jooqsimple/model/PrettyPrinter.java
package jooqsimple.model;

import org.jooq.*;
import org.jooq.conf.*;
import org.jooq.impl.*;

public class PrettyPrinter implements ExecuteListener {

    @Override
    public void executeStart(ExecuteContext ctx) {
        DSLContext create = DSL.using(ctx.dialect(), new Settings().withRenderFormatted(true));

        if (ctx.query() != null) {
            System.out.println(create.renderInlined(ctx.query()));
        } else if (ctx.routine() != null) {
            System.out.println(create.renderInlined(ctx.routine()));
        } else {
            for (Query query : ctx.batchQueries()) {
                System.out.println(create.renderInlined(query));
            }
        }
        System.out.println();
    }

}

File: src/test/kotlin/se/strindberg/jooqsimple/PrettyPrinter.kt
package se.strindberg.jooqsimple

import org.jooq.ExecuteContext
import org.jooq.ExecuteListener
import org.jooq.conf.Settings
import org.jooq.impl.DSL

class PrettyPrinter : ExecuteListener {

    override fun executeStart(ctx: ExecuteContext) {
        val create = DSL.using(ctx.dialect(), Settings().withRenderFormatted(true))

        if (ctx.query() != null) {
            println(create.renderInlined(ctx.query()))
        } else if (ctx.routine() != null) {
            println(create.renderInlined(ctx.routine()))
        } else {
            ctx.batchQueries().forEach {
                println(create.renderInlined(it))
            }
        }
        println()
    }
}
